### PR TITLE
91 create badge component

### PR DESCRIPTION
**Pull request by Atzac**
## Description

 **Badge component creation**

The Badge component contains two parameters:

- Label: Accepts String or Number, formatting the string also if it contains only numbers
- Type: primary, secondary, neutral or negative

### CODE REVIEW

File: src/components/badge/badge.test.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  return render(<IonBadge {...props} />);
+}
+
+describe('BadgeComponent', () => {
+  describe('Default', () => {
+    it('should render badge component', async () => {
+      const label = 'Projeto';
+      sut({ ...defaultBadge, label: label });
+      expect(screen.getByText(label)).toBeInTheDocument();
+    });
```
Comment (iurynogueira) [inline src/components/badge/badge.test.tsx:22]:
```suggestion
    it('should render badge component', async () => {
      sut();
      expect(screen.getByText(defaultBadge.label)).toBeInTheDocument();
    });
```

File: src/components/badge/badge.test.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  type: 'primary',
+};
+
+function sut(props: BadgeProps = defaultBadge) {
+  return render(<IonBadge {...props} />);
+}
+
+describe('BadgeComponent', () => {
+  describe('Default', () => {
+    it('should render badge component', async () => {
```
Comment (iurynogueira) [inline src/components/badge/badge.test.tsx:18]: create a new spect to check if primary is default type

File: src/components/badge/badge.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+export type BadgeProps = {
+  label: LabelType;
+  type?: BadgeType;
+} & StitchesBadgeProps;
+
+const IonBadge = ({ label, type = 'primary' }: BadgeProps) => {
+  const [badgeValue, setBadgeValue] = useState<LabelType>(0);
+
+  function formatLabel(badgeLabel: LabelType): LabelType {
```
Comment (iurynogueira) [inline src/components/badge/badge.tsx:18]: arrow function

**Review by iurynogueira — CHANGES_REQUESTED** [review]
Great! :D

File: src/components/badge/badge.test.tsx [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+const defaultBadge: BadgeProps = {
+  label: 'Badge',
+  type: 'primary',
+};
+
+function sut(props: BadgeProps = defaultBadge) {
+  return render(<IonBadge {...props} />);
+}
+
```
[the remark this answers is not in the document]
Comment (Atzac) [reply src/components/badge/badge.test.tsx:15]: Very nice!

**Review by wermesonsl — APPROVED** [review]

**Review by iurynogueira — CHANGES_REQUESTED** [review]

**Review by iurynogueira — APPROVED** [review]

**Pull request merged**